### PR TITLE
Yul optimizer tests

### PR DESCRIPTION
**Pull request by chriseth**
Depends on https://github.com/ethereum/solidity/pull/5179

### CODE REVIEW

**Comment by chriseth** [comment]
Fixed test registration and rebased onto new #5179

File: test/tools/CMakeLists.txt [hunk 1]
```diff
@@ -3,5 +3,5 @@ target_link_libraries(solfuzzer PRIVATE libsolc evmasm ${Boost_PROGRAM_OPTIONS_L
 
 add_executable(isoltest isoltest.cpp ../Options.cpp ../Common.cpp ../libsolidity/TestCase.cpp ../libsolidity/SyntaxTest.cpp
         ../libsolidity/AnalysisFramework.cpp ../libsolidity/SolidityExecutionFramework.cpp ../ExecutionFramework.cpp
-        ../RPCSession.cpp ../libsolidity/ASTJSONTest.cpp)
+		../RPCSession.cpp ../libsolidity/ASTJSONTest.cpp ../libjulia/YulOptimizerTest.cpp)
```
Comment (leonardoalt) [inline test/tools/CMakeLists.txt]: Extra tab

**Comment by leonardoalt** [comment]
How can you see in the tests logs that the tests actually ran?

**Comment by chriseth** [comment]
It seems you can't... I'll add the test results to the artifacts, then we might be able to download them.

**Comment by leonardoalt** [comment]
Or you could make a couple tests fail

**Comment by chriseth** [comment]
Asked circle to store the test results as artifacts.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**